akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -114,13 +170,20 @@ private[akka] class MergeHub[T](perProducerBufferSize: Int)
     private val queue = new AbstractNodeQueue[Event] {}
     @volatile private[this] var needWakeup = false
     @volatile private[this] var shuttingDown = false
+    private[this] var draining = false
```

Comment:
Needs to be `@volatile` since it is used from the `Sink`s which are running on other threads than this stage.